### ISSUE TITLE
[Bug]: composition - CORS issues loading stories/index .json

### ISSUE THREAD

**Issue by ndelangen** [storybookjs/storybook#21500]
### Describe the bug

@yannbf reported this during QA:

> I was not able to set up composition for https://storybook.chakra-ui.com/ with the following error:
Access to fetch at 'https://storybook.chakra-ui.com/index.json' from origin 'http://localhost:6006/' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
> 
> I tested this in Storybook 6 and it worked correctly.

### To Reproduce

Add a ref to `code/ui/.storybook/main.ts` to`https://storybook.chakra-ui.com`

These should be no CORS warning in the browser.

---

As a bit of a side-note, perhaps we could fetch the stories json file from node, and actually pass the resulting JSON into the stories field for the ref.. making the fetch in the browser no longer needed at all.

I do wonder how this worked in 6.5 then, 1 theory is that we try to fetch without CORS first, and then fallback to trying with CORS after.. probably for auth purposes?

**Comment by tmeasday** [comment]
OK, the `http-server` one was a furphy, I was running `npx http-server --cors storybook-static`, when I needed to run `npx http-server storybook-static --cors`.

**Comment by tmeasday** [comment]
In the case of Chakra, the issue is that fetch is throwing when it 404s, rather than returning a `{ ok: false }` response. This seems to be changed behaviour in fetch that we don't handle properly, going to fix.

**Comment by shilman** [comment]
Yowza!! I just released https://github.com/storybookjs/storybook/releases/tag/v7.0.0-rc.6 containing PR #21682 that references this issue. Upgrade today to the `@next` NPM tag to try it out!

```
npx sb@next upgrade --prerelease
```

**Comment by ajenkins-mparticle** [comment]
If anybody lands on this bug because of their googling and their issue isn't specifically CORS, I ran into this because I had an empty `.stories` file
